### ISSUE TITLE
[libigl[cgal]] build failure

### ISSUE THREAD

**Issue by Nosenzor** [microsoft/vcpkg#30332]
**Host Environment**

- OS: Windows 10 x64 (and also on  Mac ARM)
- Compiler: revision

**To Reproduce**

`./vcpkg install libigl[cgal]`

**Failure logs**

- Cut and paste the appropriate build messages from the console output.
- Attach ***every failure log file*** mentioned in the console output.
Make Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" C:/Users/Shadow/CLionProjects/PyH3Dxx/vcpkg/buildtrees/libigl/src/d881f96777-73695629ab.clean -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=C:/Users/Shadow/CLionProjects/PyH3Dxx/vcpkg/packages/libigl_x64-windows/debug -DFETCHCONTENT_FULLY_DISCONNECTED=ON -DLIBIGL_WITH_EMBREE=OFF -DLIBIGL_WITH_OPENGL=OFF -DLIBIGL_WITH_OPENGL_GLFW=OFF -DLIBIGL_WITH_OPENGL_GLFW_IMGUI=OFF -DLIBIGL_WITH_XML=OFF -DLIBIGL_WITH_CGAL=ON -DLIBIGL_BUILD_PYTHON=OFF -DLIBIGL_EXPORT_TARGETS=ON -DLIBIGL_USE_STATIC_LIBRARY=OFF -DLIBIGL_WITH_COMISO=OFF -DLIBIGL_WITH_TETGEN=OFF -DLIBIGL_WITH_TRIANGLE=OFF -DLIBIGL_WITH_PREDICATES=OFF -DLIBIGL_BUILD_TUTORIALS=OFF -DLIBIGL_WITH_PNG=OFF -DLIBIGL_BUILD_TESTS=OFF -DPYTHON_EXECUTABLE= "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -DBUILD_SHARED_LIBS=ON -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/Shadow/CLionProjects/PyH3Dxx/vcpkg/scripts/toolchains/windows.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=v143 -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=C:/Users/Shadow/CLionProjects/PyH3Dxx/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=C:/Users/Shadow/CLionProjects/PyH3Dxx/vcpkg -D_VCPKG_INSTALLED_DIR=C:/Users/Shadow/CLionProjects/PyH3Dxx/cmake-build-release/vcpkg_installed -DVCPKG_MANIFEST_INSTALL=OFF
    Working Directory: C:/Users/Shadow/CLionProjects/PyH3Dxx/vcpkg/buildtrees/libigl/x64-windows-dbg
    Error code: 1
    See logs for more information:
      C:\Users\Shadow\CLionProjects\PyH3Dxx\vcpkg\buildtrees\libigl\config-x64-windows-dbg-CMakeCache.txt.log
      C:\Users\Shadow\CLionProjects\PyH3Dxx\vcpkg\buildtrees\libigl\config-x64-windows-dbg-out.log
      C:\Users\Shadow\CLionProjects\PyH3Dxx\vcpkg\buildtrees\libigl\config-x64-windows-dbg-err.log

**Additional context**

Add any other context about the
[patch-x64-windows-2-out.log](https://github.com/microsoft/vcpkg/files/11027361/patch-x64-windows-2-out.log)
[patch-x64-windows-0-err.log](https://github.com/microsoft/vcpkg/files/11027362/patch-x64-windows-0-err.log)
[patch-x64-windows-0-out.log](https://github.com/microsoft/vcpkg/files/11027364/patch-x64-windows-0-out.log)
[patch-x64-windows-1-err.log](https://github.com/microsoft/vcpkg/files/11027365/patch-x64-windows-1-err.log)
[patch-x64-windows-1-out.log](https://github.com/microsoft/vcpkg/files/11027366/patch-x64-windows-1-out.log)
[patch-x64-windows-2-err.log](https://github.com/microsoft/vcpkg/files/11027367/patch-x64-windows-2-err.log)
 problem here, such as what you have already tried to resolve the issue.
[config-x64-windows-dbg-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/11027357/config-x64-windows-dbg-CMakeCache.txt.log)
[config-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/11027358/config-x64-windows-dbg-err.log)
[config-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/11027359/config-x64-windows-dbg-out.log)

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. Could you please confirm the folder _[VCPKG_PATH]/buildtrees/libigl/src/d881f96777-73695629ab.clean/external/cgal_ have files of `cgal`? They are download by the following codes in **cmake/LibiglDownloadExternal.cmake**:
```
## CGAL
function(igl_download_cgal)
	igl_download_project(cgal
		GIT_REPOSITORY https://github.com/CGAL/cgal.git
		GIT_TAG        f7c3c8212b56c0d6dae63787efc99093f4383415
	)
endfunction()
```

**Comment by dg0yt** [comment]
> They are download by the following codes in **cmake/LibiglDownloadExternal.cmake**:

@LilyWangLL Shouldn't cgal come from vcpkg?

**Comment by LilyWangLL** [comment]
> Shouldn't cgal come from vcpkg?

Now `cgal` used by `libigl` is downloaded by itself.

**Comment by dg0yt** [comment]
> > Shouldn't cgal come from vcpkg?
> 
> Now `cgal` used by `libigl` is downloaded by itself.

This is bad. Even more with `libigl[cgal]` depending `cgal`.

**Comment by LilyWangLL** [comment]
> > > Shouldn't cgal come from vcpkg?
> > 
> > 
> > Now `cgal` used by `libigl` is downloaded by itself.
> 
> This is bad. Even more with `libigl[cgal]` depending `cgal`.

I agree with this and I will fix it. I also want confirm the reason of installed failed.

**Comment by LilyWangLL** [comment]
> CMake Error at cmake/DownloadProject.cmake:179 (message):
>   Build step for cgal failed: 1
> Call Stack (most recent call first):
>   cmake/LibiglDownloadExternal.cmake:25 (download_project)
>   cmake/LibiglDownloadExternal.cmake:37 (igl_download_project_aux)
>   cmake/LibiglDownloadExternal.cmake:44 (igl_download_project)
>   cmake/libigl.cmake:226 (igl_download_cgal)
>   CMakeLists.txt:59 (include)

The error log in **config-x64-windows-dbg-err.log**.

**Comment by applicate2628** [comment]
I have successful build with libigl[*] x64-windows but it fails at x64-windows-static for libigl[opengl] or libigl[cgal]. libigl[triangle,tetgen,xml] is ok.

**Comment by dg0yt** [comment]
libigl was updated two weeks ago (https://github.com/microsoft/vcpkg/pull/27963). And I opened #30735 now. If the issue with feature cgal still exists, add the error log.

**Comment by dg0yt** [comment]
libigl[cgal] builds at least with #30735: https://dev.azure.com/vcpkg/public/_build/results?buildId=87878&view=results

**Comment by Nosenzor** [comment]
It makes sense that it doesn't build on uwp or win arm64 since libgl is not available on those platforms.

**Comment by dg0yt** [comment]
> It makes sense that it doesn't build on uwp or win arm64 since libgl is not available on those platforms.

This issue is not about uwp.
For arm64-osx, some features may build: https://github.com/libigl/libigl/issues/1686.

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. Now `libigl[cgal]` could install succeed on the latest vcpkg, so I am closing this issue for now.
